From 8256e2cc53934fce715a0b1b6a3c52dce460319c Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Tue, 23 Sep 2025 14:00:09 +0800
Subject: [PATCH 093/105] wifi: mt76: mt7996: enable Restricted TWT bit in EHT
 MAC cap

Enable Restricted TWT Support bit in EHT MAC Capabilities.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>
---
 mt7996/init.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mt7996/init.c b/mt7996/init.c
index e97179492..b4f247931 100644
--- a/mt7996/init.c
+++ b/mt7996/init.c
@@ -1821,7 +1821,8 @@ mt7996_init_eht_caps(struct mt7996_phy *phy, enum nl80211_band band,
 	if (iftype == NL80211_IFTYPE_AP)
 		eht_cap_elem->mac_cap_info[0] |=
 			IEEE80211_EHT_MAC_CAP0_SCS_TRAFFIC_DESC |
-			IEEE80211_EHT_MAC_CAP0_EPCS_PRIO_ACCESS;
+			IEEE80211_EHT_MAC_CAP0_EPCS_PRIO_ACCESS |
+			IEEE80211_EHT_MAC_CAP0_RESTRICTED_TWT;
 
 	eht_cap_elem->mac_cap_info[1] |=
 		IEEE80211_EHT_MAC_CAP1_MAX_AMPDU_LEN_MASK;
-- 
2.45.2

